src/@components/@common/MenuBar/index.tsx

```diff
@@ -30,9 +30,15 @@ export default function MenuBar(props: MenuBarProps) {
     closeMenuBar();
   };
 
+  const openElement = useOutClickCloser({
+    handleOutClickCloser: () => {
+      closeMenuBar();
+    },
+  });
```

Comment:
아 이제 아웃클릭 코드들 보니까 되게 신기하게 짰네???

먼가 더 보편적인 방법으로는, 

```typescript
export default function useOutClickCloser(handleOutClickCloser: () => void;) {
```

이렇게 props 인자가 아닌, 콜백함수 인자로 받아와서

```suggestion
  const outClickRef = useOutClick(closeMenuBar);  //ref 임을 명시해주는 변수명이 더 좋을 것 같아!
```

나라면 이런 식으로 사용할 것 같다 !!

객체로 감싸서 넘겨주는 게, 그 "객체 안에 내용"만 바뀌면 "바뀐 줄 모르는" 현상이 일어나서
저렇게 인자 인수 형태로 넘겨주는 게 더 좋지 않을까 싶다!

이 외에도, 컴포넌트에서도 prop로 넘겨줄 때 객체형보다는 최대한 원시 자료형으로 객체를 분리해서 하나하나 넘겨주는 게 더 정확하고 효율적으로 렌더링(최적화) 된다고 하더라고!

Reply:
오홍 그렇구나😮😮 해보고 있는데ㅓ 콜백함수 인자로 전달해주는 부분을 어떻게 해야할지 잘 모르게써요!..!

Reply:
`useOutClick`는 `useOutClickCloser` 요거를 바꿔서 쓴걸까요?-?